src/NHibernate/Impl/AbstractSessionImpl.cs

```diff
@@ -218,19 +256,24 @@ public virtual IQuery GetNamedSQLQuery(string name)
 			}
 		}
 
+		// 6.0 TODO: remove virtual from below properties.
+		/// <inheritdoc />
+		public virtual ConnectionManager ConnectionManager { get; protected set; }
```

Comment:
If we were willing to fix serialization for stateless, we would need to serialize it. Not having it serializable is a bug. Since we need to serialize it for `SessionImpl`, better reflect that fact. This anyway changes nothing for serializable, since it cannot be be deserialize currently.